api/envoy/extensions/rbac/matchers/upstream_port/v3/upstream_port_matcher.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package envoy.extensions.rbac.matchers.upstream_port.v3;
+
+import "envoy/type/v3/range.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.rbac.matchers.upstream_port.v3";
+option java_outer_classname = "UpstreamPortMatcherProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: RBAC upstream port range matcher plugin]
+// [#extension: envoy.rbac.matchers.upstream_port]
+
+// This is configuration for matching upstream port.
+message UpstreamPortMatcher {
+  // A Port range that will be used to match the upstream port.
+  // This matcher requires a filter in the chain to have saved the upstream address(es) in the
+  // filter state before the matcher is executed by RBAC filter. The state should be saved with key
+  // `filter_state_key.set_object` (See
```

Comment:
This doesn't seem a very descriptive key? See my comment below. We should clarify what the file state object type is I think first.

Reply:
@htuch I redid the key and the container struct based on suggestions from you and @mattklein123 . PTAL/